### ISSUE TITLE
[Feature Request] `format` should accept undefined as a valid value

### ISSUE THREAD

**Issue by Chaphasilor** [Olian04/better-logging#51]
**Describe the bug**

I want to add a custom formatter, but only in production mode. What I tried is this: 

```node
format: process.env.environment !== `production` ? undefined : ctx => `...`,
```

This works in production mode, but not in any other mode, because `undefined` isn't a function.

```
TypeError: config.format is not a function   
    at Object.exports.formatMessage (...\node_modules\better-logging\dist\lib\formatMessage.js:44:37)     
    at ...\node_modules\better-logging\dist\lib\decorateObject.js:21:58
```

I'm guessing this is due to the use of something like `Object.entries()`, which will yield the key `format` with value `undefined` if format is specified as such, instead of omitting it, like it does when it's *actually* not defined.

**Expected behavior**

A simple error handling to check if the value of `format` (or any other option, for that matter) is `undefined`, and if it is, handle it the exact same way as if it we *actually* undefined :)

**Desktop (please complete the following information):**
 - Windows 10, Node.js 14.15.1

**Additional context**

The package is great! :D

**Comment by Olian04** [comment]
Thank you for opening an issue!

The behaviour your experiencing is by design, and I've yet to decide if I should concider it a flawed design or a strict one.

For the time being, consider doing the following instead:

```js
const commonConf = { /* Some common configuration */ }
const prodConf = {
   ...commonConf,
   format: ctx => /*Some format string*/
}
const devConf = {
  ...commonConf,
}
const conf = process.env.environment === `production` ? prodConf : devConf; 

betterLogging(console, conf);
```

**Comment by Chaphasilor** [comment]
Hey @Olian04, thanks for getting back to me :D 

I know how to work around the limitation, I'd just prefer it if I didn't have to :)

Why do you think treating `undefined` this way is a good choice? What are the benefits? :)  
I thought maybe it's about preventing missing logs, but if it acts as if nothing was provided, it would simply use the default formatter...

Would you care to explain it further please? :)

**Comment by Chaphasilor** [comment]
Sooo, have you thought about it? \^\^

**Comment by Olian04** [comment]
@Chaphasilor I think I've decided that I don't have any strong opinions either way. So might as well go with your idea. I'll have a look at a clean implementation soon.

**Comment by Olian04** [comment]
Might implement this as part of the helper library. https://github.com/Olian04/typescript-helpers/issues/1

**Comment by Chaphasilor** [comment]
> I think I've decided that I don't have any strong opinions either way. So might as well go with your idea. I'll have a look at a clean implementation soon.

great to hear that!

> Might implement this as part of the helper library. Olian04/typescript-helpers#1

hmm, I actually don't use typescript at all :/

**Comment by Olian04** [comment]
> > Might implement this as part of the helper library. [Olian04/typescript-helpers#1](https://github.com/Olian04/typescript-helpers/issues/1)
> 
> hmm, I actually don't use typescript at all :/

That was just a comment for my self. As a reminder / todo.

**Comment by Olian04** [comment]
Implemented in release [4.5.0](https://github.com/Olian04/better-logging/releases/tag/4.5.0)

**Comment by Chaphasilor** [comment]
Works like a charm, sweet! :D

Thanks a ton <3
